### DISCUSSION TITLE
How to change the style of a text3d object ?

### DISCUSSION THREAD

**Discussion by GMezWheel** [zauberzeug/nicegui#2141] [Q&A] [answered]
### Question

I want to dynamically change the background color of a text3d object in my ui.scene.

For a ring object, I used item.material("#FFA500").

But this doesn't work for a text3d object.

It seems that I need to modify the object's style.

How can I do this ?

**Comment by falkoschindler** [comment, marked as answer]
Hi @GMezWheel!
The support for styling 3D text is currently rather limited and could certainly be improved. But as a workaround the following JavaScript call might be useful:
```py
with ui.scene().classes('w-full h-64') as scene:
    text = scene.text3d('3D', 'background: silver').with_name('t')

ui.button('change color', on_click=lambda: ui.run_javascript(f'''
    getElement({scene.id}).scene.getObjectByName('{text.name}').element.style.backgroundColor = 'snow';
'''))
```

**Comment by GMezWheel** [comment]
Hi @falkoschindler,

It's work.
Thanks a lot...
